### PR TITLE
[BUG FIX] Opening and closing RollCalls and checks for question ids and vote ids in Be2 Scala

### PR DESCRIPTION
**Pull request by Ajkunas**
Fixed bug mentioned in issue #967 concerning the RollCall validators and RollCall handlers
Added checks for question ids and vote ids in the Election validators

### CODE REVIEW

**Review by Tesa27 — APPROVED** [review]
Karate confirms the check for election ids and vote ids, the extra checks for closing a closed roll call and opening an opened roll call also seem to work. To be done in an upcoming PR is to check that closing an unopened roll call and opening a non created roll call should be considered as invalid by the backend too. Good job overall!

**Review by lourkeur — APPROVED** [review]
Reviewed off-line due to web interface issues, LGTM

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala [hunk 1]
```diff
@@ -51,6 +51,10 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
 
         val sender: PublicKey = message.sender
 
+        //checks if the question ids are valid wrt protocol
+        def validateQuestionId(questions: List[ElectionQuestion], election_id: Hash): Boolean =
+          questions.forall(q => q.id == Hash.fromStrings("Question", election_id.toString, q.question))
```
Comment (RaulinN) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala:56]: Would be nice to have this as a constant in one of the related classes. Same for line 157 with "Vote" =)

**Review by RaulinN — APPROVED** [review]
Nice job! I was confused in some commits because it looks like there are changes than Lourkeur did in a previous PR 🤔. Other than that, nice work

**Pull request merged**